### PR TITLE
MINOR: enable reassign_partitions_test.py for kraft

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

File: tests/kafkatest/tests/core/reassign_partitions_test.py [hunk 1]
```diff
@@ -130,8 +133,9 @@ def move_start_offset(self):
 
     @cluster(num_nodes=8)
     @matrix(bounce_brokers=[True, False],
-            reassign_from_offset_zero=[True, False])
-    def test_reassign_partitions(self, bounce_brokers, reassign_from_offset_zero):
+            reassign_from_offset_zero=[True, False],
+            metadata_quorum=quorum.all_non_upgrade)
+    def test_reassign_partitions(self, bounce_brokers, reassign_from_offset_zero, metadata_quorum):
```
Comment (ijuma) [inline tests/kafkatest/tests/core/reassign_partitions_test.py:138]: How is `metadata_quorum` used?

**Comment by ijuma** [comment]
cc @rondagostino

**Comment by rondagostino** [comment]
LGTM

> How is `metadata_quorum` used?

`KafkaService` uses the value of `metadata_quorum` to determine what type of metadata quorum the test should use, and there are a bunch of things that have to happen if the value indicates co-located or remote KRaft (broker config changes, potentially creating/configuring/starting a remote controller quorum).  The actual test code itself (`test_reassign_partitions()` in this case) doesn't use the value directly, but we have to declare the parameter in order to inject the value.

`KafkaService` generally uses the injected value by invoking either:

1. `self.quorum_info = quorum.ServiceQuorumInfo(self, context)` to get the general quorum information for the service (whether it is using ZooKeeper, co-located KRaft, or remote KRaft)

-- or --

2. `quorum.NodeQuorumInfo(self.quorum_info, node)` to get the specific details for a particular node (i.e. what it's `broker.roles` value should be, since it is potentially different across the different nodes of a Kafka cluster when using co-located KRaft)

**Comment by rondagostino** [comment]
> `KafkaService` uses the value of `metadata_quorum` to determine what type of metadata quorum the test should use

On more thing: the code assumes `quorum.ZK` if no value was injected so that existing test definitions continue to work.  See https://github.com/apache/kafka/blob/trunk/tests/kafkatest/services/kafka/quorum.py#L33-L40 for details.

**Comment by ijuma** [comment]
Thanks for the explanation @rondagostino.

**Review by ijuma — APPROVED** [review]
LGTM, thanks @cmccabe.

**Pull request merged**